src/headers/mod.rs

```diff
@@ -0,0 +1,86 @@
+use async_trait::async_trait;
+use reqwest::header::HeaderMap;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+mod lnurl_auth_jwt;
+pub use lnurl_auth_jwt::LnurlAuthJwt;
+
+/// Errors around providing headers for each VSS request.
+#[derive(Debug)]
+pub enum VssHeaderProviderError {
+	/// Invalid data was encountered.
+	InvalidData {
+		/// The error message.
+		error: String,
+	},
+	/// An external request failed.
+	RequestError {
+		/// The error message.
+		error: String,
+	},
+	/// An application-level error occurred specific to the header provider functionality.
+	ApplicationError {
+		/// The error message.
+		error: String,
+	},
+}
+
+impl Display for VssHeaderProviderError {
+	fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+		match self {
+			Self::InvalidData { error } => {
+				write!(f, "invalid data: {}", error)
+			}
+			Self::RequestError { error } => {
+				write!(f, "error making request: {}", error)
+			}
+			Self::ApplicationError { error } => {
+				write!(f, "application error providing headers: {}", error)
+			}
+		}
+	}
+}
+
+impl Error for VssHeaderProviderError {}
+
+/// Defines a trait around how headers are provided for each VSS request.
+#[async_trait]
+pub trait VssHeaderProvider {
+	/// Returns the HTTP headers to be used for a VSS request.
+	/// This method is called on each request, and should likely perform some form of caching.
+	async fn get_headers(&self) -> Result<HashMap<String, String>, VssHeaderProviderError>;
```

Comment:
I was adding this, but the trait does need to remain object-safe since we use `dyn VssHeaderProvider`, so we can't use generics here to provide the structured request. The binary serialization of the body can be provided though specifically for request signing. Shall I add that?

Reply:
Done.